components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,169 @@
+import { Dropdown } from "components";
+import { grey800, white } from "constants/colors";
+import { addOpacityToHsl } from "helpers";
+import { usePaginationContext } from "hooks";
+import PreviousPage from "public/assets/icons/left-chevron.svg";
+import NextPage from "public/assets/icons/right-chevron.svg";
+import styled from "styled-components";
+import { DropdownItemT, PaginateForT } from "types";
+
+export interface Props {
+  paginateFor: PaginateForT;
+  numberOfEntries: number;
+}
+
+export function Pagination({ paginateFor, numberOfEntries }: Props) {
+  const { pageStates, goToPage, nextPage, previousPage, firstPage, lastPage, setResultsPerPage } =
+    usePaginationContext();
+
+  const { pageNumber, resultsPerPage } = pageStates[paginateFor];
+  const numberOfPages = Math.ceil(numberOfEntries / resultsPerPage);
+  const lastPageNumber = numberOfPages;
+  const defaultNumberOfButtons = 4;
+  const hasMorePagesThanButtons = numberOfPages >= defaultNumberOfButtons;
+  const showFirstButton = hasMorePagesThanButtons;
+  const showLastButton = hasMorePagesThanButtons;
+  const numberOfButtons = hasMorePagesThanButtons ? defaultNumberOfButtons : numberOfPages;
+  const numbersPastMax = pageNumber - numberOfButtons;
+
+  const buttonNumbers = makeButtonNumbers();
+
+  function makeButtonNumbers() {
+    if (!hasMorePagesThanButtons) {
+      return Array.from({ length: numberOfPages }, (_, i) => i + 1);
+    }
+
+    const isLastPastNumbers = 2 * numberOfButtons + numbersPastMax >= numberOfPages;
+
+    if (isLastPastNumbers) {
+      return Array.from({ length: numberOfButtons }, (_, i) => lastPageNumber - numberOfButtons + i);
+    }
+
+    return Array.from({ length: numberOfButtons }, (_, i) => i + 2).map((number) => {
+      if (numbersPastMax > 0) {
+        return number + numbersPastMax;
+      }
+      return number;
+    });
+  }
+
+  const _goToPage = (page: number) => goToPage(paginateFor, page);
+  const _nextPage = () => nextPage(paginateFor);
+  const _previousPage = () => previousPage(paginateFor);
+  const _firstPage = () => firstPage(paginateFor);
+  const _lastPage = () => lastPage(paginateFor, lastPageNumber);
+  const _setResultsPerPage = (resultsPerPage: number) => setResultsPerPage(paginateFor, resultsPerPage);
```

Comment:
you could put this in the pagination context, same way as we did the error context